Frama-C complains:

[kernel] user error: Incompatible declaration for __switch_to:
                     different type constructors: void vs. struct task_struct
                     First declaration was at  src-linux-3.17.1/arch/um/kernel/process.c:80
                     Current declaration is at src-linux-3.17.1/include/asm-generic/switch_to.h:22

--- a/arch/um/kernel/process.c
+++ b/arch/um/kernel/process.c
@@ -22,6 +22,7 @@
 #include <asm/current.h>
 #include <asm/pgtable.h>
 #include <asm/mmu_context.h>
+#include <asm/switch_to.h>
 #include <asm/uaccess.h>
 #include <as-layout.h>
 #include <kern_util.h>
@@ -77,7 +78,7 @@ static inline void set_current(struct ta
 
 extern void arch_switch_to(struct task_struct *to);
 
-void *__switch_to(struct task_struct *from, struct task_struct *to)
+struct task_struct *__switch_to(struct task_struct *from, struct task_struct *to)
 {
 	to->thread.prev_sched = from;
 	set_current(to);
